src/itf/context-dep-itf.h

```diff
@@ -60,12 +60,36 @@ class ContextDependencyInterface {
   virtual bool Compute(const std::vector<int32> &phoneseq, int32 pdf_class,
                        int32 *pdf_id) const = 0;
 
+  /// This function outputs information about what possible pdf-ids can
+  /// be generated for HMM-states; it covers the general case where
+  /// the self-loop pdf-class may be different from the forward-transition
+  /// pdf-class, so we are asking not about the set of possible pdf-ids
+  /// for a given (phone, pdf-class), but the set of possible ordered pairs
+  /// (forward-transition-pdf, self-loop-pdf) for a given (phone,
+  /// forward-transition-pdf-class, self-loop-pdf-class).
+  /// Note: 'phones' is a list of integer ids of phones, and
+  /// 'pdf-class-pairs', indexed by phone, is a list of pairs
+  /// (forward-transition-pdf-class, self-loop-pdf-class) that we can have for
+  /// that phone.
+  /// The output 'pdf_info' is indexed first by phone and then by the
+  /// same index that indexes each element of 'pdf_class_pairs',
+  /// and tells us for each pair in 'pdf_class_pairs', what is the
+  /// list of possible (forward-transition-pdf-id, self-loop-pdf-id) that
+  /// we can have.
+  /// This is less efficient than the other version of GetPdfInfo().
+  virtual void GetPdfInfo(
+      const std::vector<int32> &phones,
+      const std::vector<std::vector<std::pair<int32, int32> > > &pdf_class_pairs,
+      std::vector<std::vector<std::vector<std::pair<int32, int32> > > > *pdf_info)
+      const = 0;
+
   /// GetPdfInfo returns a vector indexed by pdf-id, saying for each pdf which
   /// pairs of (phone, pdf-class) it can correspond to.  (Usually just one).
   /// c.f. hmm/hmm-topology.h for meaning of pdf-class.
-  virtual void GetPdfInfo(const std::vector<int32> &phones,  // list of phones
```

Comment:
Clarify that this is the old, simpler interface of GetPdfInfo(), and that this one can only be called if the HmmTopology object's IsHmm() function call returns true.  This should probably be above the other one- more readable that way.